adobe/memory.hpp

```diff
@@ -364,9 +364,9 @@ constexpr T* construct_at(T* p, Args&&... args) {
 #else
 
 template <class T, class... Args>
-constexpr T* construct_at(T* p, Args&&... args) {
```

Comment:
I don't even know what I was thinking with this code -
Having the conditional in the header is asking for ODR violations. I would suggest starting at line [356-370] be replaced with:
```cpp
template <class T, class... Args>
T* construct_at(T* p, Args&&... args) {
    return ::new (static_cast<void*>(p)) T(std::forward<Args>(args)...);
}
```
From [cppreference](https://en.cppreference.com/w/cpp/memory/construct_at). The constexpr with the cast isn't doing anything, and I'm surprised it doesn't generate a warning. Clearly, the C++20 implementation never compiled; your suggestion loses the constexpr-ness of the standard call. Let's keep it simple and when we require C++20 we can deprecate this instance.